suse-buildsystem.sh

```diff
@@ -7,3 +7,31 @@ export QT_HASH_SEED=0
 export PERL_HASH_SEED=42
 export PYTHONHASHSEED=0
 export FORCE_SOURCE_DATE=1 # for texlive to use SOURCE_DATE_EPOCH
+
+( if test -f /.buildenv && test "Y" = "$(rpm --eval '%disable_obs_set_source_date_epoch')"; then
```

Comment:
The code says to do this when `%disable_obs_set_source_date_epoch = Y` - that seems wrong.

I think, we should keep this "adding of release" off by default as it will break build-compare and build-tree-pruning for a large number of packages that already use `SOURCE_DATE_EPOCH` as timestamp.

So how about `if test -f /.buildenv && test "Y" = "$(rpm --eval '%obs_set_source_date_epoch')"`

Reply:
Thx, I need to fix the test.

Where do you expect default off to help? Either we publish and thus need this enabled or we do not do much build-compare based dependency-tree rebuild pruning because the number of packages are small because it is not a full distribution.

In either case the real solution is to fix upstream to not put dates in the output. Depending on the case if something is really needed there is usually something that is more meaningful. E.g. you could put the major version of the software instead. And for workarounds in spec files set SDE to 1 for a subprocess where you ensured it is not used for a file mtime. Either of these will result in build-compare still working.

Reply:
Both BUILD_CHANGELOG_TIMESTAMP and git commit mtime are also still available inside the spec file.

Reply:
For ease of use I leave it as BROKEN_SOURCE_DATE_EPOCH in the environment.